src/metax_api/tests/api/rest/base/views/datasets/write.py

```diff
@@ -253,6 +265,115 @@ def test_anonymous_users_access_to_draft_datasets(self):
+    def test_endusers_can_update_draft_datasets(self):
+        ''' End user should be able to update only his/her drafts '''
+        # Test update access as end user
+        self._use_http_authorization(method='bearer', token=self.token)
+
+        data1 = self.client.get('/rest/datasets/1').data # published
+        response = self.client.patch('/rest/datasets/1', data1, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data2 = self.client.get('/rest/datasets/2').data # end users own draft
+        response = self.client.patch('/rest/datasets/2', data2, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data3 = self.client.get('/rest/datasets/3').data # someone elses draft
+        response = self.client.patch('/rest/datasets/3', data3, format="json")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND, response.data)
+
+        # try put
+        response = self.client.put('/rest/datasets/1', data1, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        response = self.client.put('/rest/datasets/2', data2, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        response = self.client.put('/rest/datasets/3', data3, format="json")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND, response.data)
+
+        # test for multiple datasets
+
+        response = self.client.patch('/rest/datasets', [data1, data2, data3], format="json")
+        owners = [cr['object']['metadata_provider_user'] for cr in response.data['success']]
+        self.assertEqual('#### Some owner who is not you ####' not in owners, True, response.data)
+
+        response = self.client.put('/rest/datasets', [data1, data2, data3], format="json")
+        owners = [cr['object']['metadata_provider_user'] for cr in response.data['success']]
+        self.assertEqual('#### Some owner who is not you ####' not in owners, True, response.data)
+
+    def test_service_users_can_update_draft_datasets(self):
+        ''' Service users should get all data '''
+        # test access as a service-user
+        self._use_http_authorization(method='basic', username='metax')
+
+        data1 = self.client.get('/rest/datasets/1').data # published
+        response = self.client.patch('/rest/datasets/1', data1, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data2 = self.client.get('/rest/datasets/2').data # draft
+        response = self.client.patch('/rest/datasets/2', data2, format="json")
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        # Dataset drafts should be able to be updated by service users (service is responsible that
+        # their current user in e.g. Qvain is allowed to access the dataset)
```

Comment:
Comment probably belongs to method's docstring? (the '''comment''' at the beginning of the method)